siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+]
+
+streamlit.set_page_config(page_title="SiliconCompiler",
+                          page_icon=Image.open(sc_logo_path),
+                          layout="wide",
+                          menu_items={
+                              "Get help": "https://docs.siliconcompiler.com/",
+                              "Report a Bug": '''https://github.com/siliconcomp
+                              iler/siliconcompiler/issues''',
+                              "About": "\n\n".join(sc_about)
+                          })
+
+
+def modify_logs_and_reports_to_streamlit(logs_and_reports):
+    """
+    Converts the logs_and_reports found to the strucutre
+    required by streamlit_tree_select. Success is predicated on the order of
+    logs_and_reports outlined in report.get_logs_and_reports
+
+    Args:
+        logs_and_reports (list) : a list of 3-tuples with order of a path name,
+                                  folder in the subdirectory, and files in the
+                                  subdirectory
+    """
+    subsect_logs_and_reports = {}
+    if logs_and_reports == []:
+        return []
+    starting_path_name = logs_and_reports[0][0]
+    # reverse the list to start building the tree from the leaves up
+    for path_name, folders, files in reversed(logs_and_reports):
+        children = []
+        for folder in folders:
+            # assert (folder in subsect_logs_and_reports)
+            children.append(subsect_logs_and_reports[folder])
+        for file in files:
+            node = {}
+            node['label'] = file
+            # TODO: change to accomodate different operating systems
+            node['value'] = f'{path_name}/{file}'
+            children.append(node)
+        if starting_path_name == path_name:
+            return children
+        else:
+            node = {}
+            # TODO: change to accomodate different operating systems
+            folder = path_name[path_name.rfind('/') + 1:]
+            node['label'] = folder
+            node['value'] = path_name
+            node['children'] = children
+            subsect_logs_and_reports[folder] = node
+
+
+def get_nodes_and_edges(chip, node_dependencies, successful_path):
+    """
+    Returns the nodes and edges required to make a streamlit_agraph.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        node_dependencies (dict) : dictionary where the values of the keys are
+                                   the edges
+        successful_path (set) : contains all the nodes that are part of the
+                                'winning' path
+    """
+    nodes = []
+    edges = []
+    opacity = 0.2
+    width = 1
+    for step, index in node_dependencies:
+        if (step, index) in successful_path:
+            opacity = 1
+            if (step, index) in chip._get_flowgraph_exit_nodes() or \
+               (step, index) in chip._get_flowgraph_entry_nodes():
+                width = 3
```

Comment:
magic number, make this an input with a default (what is the unit?)

Reply:
I did the first part. The documentation really isn't clear on what the unit is.